src/stan/analyze/mcmc/compute_effective_sample_size.hpp

```diff
@@ -35,6 +37,30 @@ namespace analyze {
       num_draws = std::min(num_draws, sizes[chain]);
     }
 
+    if ( num_draws < 3 ) {
+      return std::numeric_limits<double>::quiet_NaN();
+    }
+
+    Eigen::VectorXd draw_val = Eigen::VectorXd::Random(num_chains);
+    for ( int chain = 0; chain < num_chains; chain++ ) {
+      Eigen::Map<const Eigen::Matrix<double, Eigen::Dynamic, 1>>
+        draw(draws[chain], sizes[chain]);
+
+      for ( int n = 0; n < num_draws; n++ ) {
+        if ( !boost::math::isfinite(draw(n)) ) {
+          return std::numeric_limits<double>::quiet_NaN();
+        }
+      }
+
+      if ( draw.isApproxToConstant(draw(0)) ) {
+        draw_val(chain) = draw(0);
+      }
+    }
+
+    if ( draw_val.isApproxToConstant(draw_val(0)) ) {
```

Comment:
Do we not want to return NaN here?  If everything is constant then the asymptotic variance, and hence ESS, isn't particularly well-defined.

Reply:
A valid point and I agree.  I was going for consistency with the original issue's reference implementation: https://github.com/stan-dev/rstan/blob/8caa8728669ae2df75167202e08dff930d4d7c64/rstan/rstan/R/monitor.R#L213

Should we over rule the consistency argument in favor of returning NaN here?